src/select/SelectDropdown.ts

```diff
@@ -57,35 +70,36 @@ const useSelectDropdown = createHook<SelectDropdownOptions, BoxHTMLProps>({
       }
     }, [visible]);
 
-    React.useEffect(
-      debounce(() => {
-        if (typehead === "") return;
+    React.useEffect(() => {
+      debounce(
+        () => {
+          if (typehead === "") return;
 
-        items.forEach(item => {
-          if (!item.ref.current) return;
-          const dataAttrValue = item.ref.current.getAttribute(
-            "data-value",
-          ) as string;
+          items.forEach(item => {
+            if (!item.ref.current) return;
+            const dataAttrValue = item.ref.current.getAttribute(
+              "data-value",
+            ) as string;
 
-          if (
-            !selected.includes(dataAttrValue) &&
-            dataAttrValue.startsWith(typehead)
-          ) {
-            setCurrentId(item.id);
-            move(item.id);
-            // if dropdown is not open directly select the item.
-            if (!visible) {
-              // remain dropdown open on setSelected
-              setSelected(dataAttrValue, true);
+            if (
+              !selected.includes(dataAttrValue) &&
+              dataAttrValue.startsWith(typehead)
+            ) {
+              setCurrentId(item.id);
+              move(item.id);
+              // if dropdown is not open directly select the item.
+              if (!visible) {
+                // remain dropdown open on setSelected
+                setSelected(dataAttrValue, true);
+              }
             }
-          }
-        });
-      }, 150),
-      [typehead, visible],
-    );
+          });
+        },
+        visible ? 150 : 400,
```

Comment:
I'm still seeing the bug with multiselect so i'm conditionally setting the debounce value.

Reply:
We can remove the debounce, since we are not going to use it in multiselect